src/mpid/ch4/netmod/ucx/ucx_win.c

```diff
@@ -284,20 +289,19 @@ int MPIDI_UCX_mpi_win_free_hook(MPIR_Win * win)
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_UCX_MPI_WIN_FREE_HOOK);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_UCX_MPI_WIN_FREE_HOOK);
 
-    if (MPIDI_WIN(win, winattr) & MPIDI_WINATTR_NM_REACHABLE) {
```

Comment:
`MPIDI_WINATTR_NM_REACHABLE` is no longer used at all in ucx, right? If so, I would expect removing the code on line 138 that setting it as well. We also won't need the `all_reachable` local variable there for the same reason.

Separately, removing the usage of `MPIDI_WINATTR_NM_REACHABLE` and adding `MPIDI_UCX_WIN(win).reachable_flag` are orthogonal, right? If so, shouldn't they separate into separate commits?

Reply:
No, `MPIDI_WINATTR_NM_REACHABLE` is still used at `MPIDI_UCX_is_reachable_target` which is checked when issuing RMA operation or flush|flush_local.

We cannot separate change of `MPIDI_WINATTR_NM_REACHABLE` and `reachable_flag`, because `MPIDI_WINATTR_NM_REACHABLE` was used to skip `mem_unmap` at `win_free` when memory is not registered. We still have to skip in such a case, thus we need `reachable_flag` to make this commit work.